### ISSUE TITLE
magit sometimes takes emacs hostage

### ISSUE THREAD

**Issue by chriselrod** [magit/magit#4862]
I use exwm, making restarting emacs rather inconvenient. Especially if I can't do so via `C-x C-c`, in which case I must power cycle.

The problem:
Sometimes, a transient magit window appears and does not go away, intercepting all keys associated with the window. `C-g`, `q`, `ESC`, etc do not close it, and I couldn't close it via `RET`/hitting enter trying to pick options either, e.g. creating a new branch named `aaaa` (can't execute commands in the mini buffer, so the choice of letters to use is limited when this take-over happens) didn't let me escape.

Moving to other windows didn't allow breaking out of it/it still intercepted, etc.

It'd be nice if there were a way to break out of this, e.g. `C-g` or `q` support for always breaking out of transients.
Or even having a means to kill all magit buffers would be nice.

**Comment by tarsius** [comment]
Have you tried with the latest `transient` snapshot from Melpa?

**Comment by chriselrod** [comment]
It happens infrequently, and unfortunately I don't have a reproducer. It's happened only twice so far, yesterday before filing the issue being the most recent example.

I should have checked which version I was on at the time, but I am currently on master with most recent commit
```
Author: Jonas Bernoulli <jonas@bernoul.li>
Date:   Sat Jan 7 16:24:09 2023 +0100

    Bump copyright years
```
Looking at the github repo
https://github.com/magit/transient
Seems I should be on main, not master...

**Comment by tarsius** [comment]
That is a fairly recent version and it contains all the relevant changes that have already been merged into `main` (the branch formerly known as `master`).

It is possible that you are not actually using the version of `transient` that you think you are using. Try using `find-function` to go to the definition of some `transient` function and verify that it takes you to the file you expected.

Of course `exwm` is special and it might be the case that there is indeed some previously unknown issue that is specific to the role it plays.

I am currently working on yet another iteration of the relevant code. I am trying to finish and merge it as soon as possible because `transient` is part of Emacs and the first pretest for Emacs 29.1 is scheduled to happen any time now.

So I suggest you wait a few days before updating. Then use that version for a while and come back to report whether you are still running into this issue.

**Comment by jmalmari** [comment]
I experience similar. Can reproduce 100 % with following:
1. magit status buffer
2. b for branch menu
3. c for create
4. when asking for an existing ref to start from, enter something invalid. "Not a valid starting-point: asdfasdf"
5. Again: b for branch menu
6. c for create
7. In minibuffer input, it feels like all shortcut keys of the popup are still active. Characters get filtered in the minibuffer line edit. The popup is forever stuck open.

GNU Emacs 28.2 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.36, cairo version 1.17.6) of 2023-01-03
Magit 20230129.555 [>= 3.3.0.50-git], Git 2.39.1, Emacs 28.2, gnu/linux

**Comment by tarsius** [comment]
I just reverted a commit to fix another regression (magit/transient#229). Turns out doing that also fixes this issue! :grinning:

Thanks for the nice reproducer.
